apps/passwd.c

```diff
@@ -47,6 +33,14 @@ static unsigned const char cov_2char[64] = {
     0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79, 0x7A
 };
 
+static const char ascii_dollar[] = { 0x24, 0x00 };
+#if defined(CHARSET_EBCDIC) && !defined(CHARSET_EBCDIC_TEST)
+static const char ebcdic_dollar[] = { 0x5B, 0x00 };
+# define dollar ebcdic_dollar
+#else
+# define dollar ascii_dollar
+#endif
```

Comment:
I prefer @dot-asm 's suggestion here.  It doesn't allow testing now but it is the better solution long term.  Does it need to be changed now?  Probably not if it doesn't get lost (which it easily might).

The ctype replacement doesn't (thus far) change anything for EBCDIC, although it does remove some conditional code.  Rather, it address a different problem where some implementations return true for some of the calls for non-seven bit ASCII.  If reverted, we'd need to include isascii tests before a lot of ctype calls.